tests/chainerx_tests/unit_tests/routines_tests/test_logic.py

```diff
@@ -205,3 +205,105 @@ def forward_xp(self, inputs, xp):
         a, = inputs
         b = xp.logical_not(a)
         return b,
+
+
+# Hack for Issue #6778
+# TO-DO(kshitij12345): Remove when issue fixed
+def cast_if_float16(xp, a):
+    if a.dtype == 'float16':
+        return xp.logical_not(a.astype(bool, True))
+    else:
+        return a.astype(bool, True)
+
+
+def logical_and_lambda(xp, a, b):
+    return xp.logical_and(a, b)
+
+
+def logical_or_lambda(xp, a, b):
+    return xp.logical_and(a, b)
```

Comment:
Ops, this should be the following?
```suggestion
    return xp.logical_or(a, b)
```

Reply:
Oh right.

Reply:
This still seems to be an issue. Could you recheck it?